test/flint/fmpz_mod-test.jl

```diff
@@ -128,6 +131,21 @@ end
   @test_throws Exception R6(R22(1))
   @test_throws Exception R2(R3(1))
   @test_throws Exception R3(R2(1))
+
+  ZZmod720,_ = residue_ring(ZZ,720)
+  ZZmodZZ720,_ = residue_ring(ZZ,ZZ(720))
+  @test is_nilpotent(ZZmod720(30))
+  @test is_nilpotent(ZZmodZZ720(30))
+  nilp720 = filter((r -> is_nilpotent(ZZmod720(r))), 0:719);
+  @test length(nilp720) == 24  # 720/(2*3*5)
+  nilp720 = filter((r -> is_nilpotent(ZZmodZZ720(r))), 0:719);
+  @test length(nilp720) == 24  # 720/(2*3*5)
+
+  # 64-bit overflow test:  **ASSUMES** Int is 64 bits
+  rr = 4*3^20
+  mm = 3*rr
+  R_overflow,_ = residue_ring(ZZ, mm)
+  @test is_nilpotent(R_overflow(rr))
```

Comment:
What exactly is this testing? Clearly it is related to `Int` overflowing, but which aspect of that / where is it coming into play here?

Reply:
The current implementation of `is_nilpotent` works correctly for this input, but an earlier version was incorrect (even though the code looked right).  I prefer to keep this test example in case someone in the future modifies the code, and mishandles integer overflow.  Maybe this should be in `nmod-test` too?